packages/ipns/src/index.ts

```diff
@@ -201,7 +266,13 @@ class DefaultIPNS implements IPNS {
   }
 
   async resolveDns (domain: string, options: ResolveDNSOptions = {}): Promise<CID> {
-    const dnslink = await resolveDnslink(domain, options)
+    const resolvers = options.resolvers ?? [
+      defaultResolver()
```

Comment:
Can this be called once and reused rather creating a new function each time the fallback is hit?